src/cpp/src/whisper_generation_config.cpp

```diff
@@ -114,6 +81,15 @@ void WhisperGenerationConfig::validate() const {
         OPENVINO_ASSERT(!language.has_value(), "Cannot specify 'language' for not multilingual model.");
         OPENVINO_ASSERT(!task.has_value(), "Cannot specify 'task' for not multilingual model.");
     }
+
+    if (is_beam_search()) {
+        OPENVINO_ASSERT(num_return_sequences == 1,
```

Comment:
is it because our Whisper API does not allow to provide multiple outputs?
If yes, than even w/o beam_search we need to assert to this condition